Admin Activity Log: generate unique id for each request #4444

kanghj opened the pull request:
Fixes #4444

damithc commented:
Can give an example of a generated ID, just to see how it would look like?

kanghj commented:
Log on app engine: Added the id before the time taken column
![log_appengine](https://cloud.githubusercontent.com/assets/2216119/10570744/ea5ff7b0-7669-11e5-89be-8fb04f90e9bc.PNG)
Log on admin activity log: Added the "id:" before the user's name
![log_teammates](https://cloud.githubusercontent.com/assets/2216119/10570748/ef1dc8c2-7669-11e5-94b1-541d519f57bc.PNG)

damithc commented:
Is that the unix time? May be convert time to a more meaningful format? like 20151019123454...

kanghj commented:
Log on app engine:
![log_appengine](https://cloud.githubusercontent.com/assets/2216119/10580944/bbf51b46-76b2-11e5-816c-c50fd5dea501.PNG)


Log in adminActivityLogPage:
![log_teammates](https://cloud.githubusercontent.com/assets/2216119/10580945/bf4bef0e-76b2-11e5-8204-add4257e4e2e.PNG)

Changed the time format to be more meaningful. Does this look better?

damithc commented:
When I'm searching for a particular log entry in the app engine console,
the log search feature might get confused by dots and treat them as
separate words. If you can confirm it's not a problem, then can keep the
dots. Otherwise remove the dots.

kanghj commented:
Removed the dots. The newer app engine log on the developer console treats it as a text search, but the older log uses regex. Although it still works correctly, I removed it just to be safe. Ready for review, @bamboo3250

bamboo3250 commented on src/main/java/teammates/common/util/ActivityLogEntry.java:
```diff
@@ -143,6 +155,8 @@ public ActivityLogEntry(String servlet, String act, AccountAttributes acc, Strin
             email = acc.email;
         }
         
+        id = googleId + time;
```
Why don't you use `formatCurrentTimeForId() ` here like the line 217 below? Likewise for the line 115.

bamboo3250 commented on src/main/java/teammates/common/util/ActivityLogEntry.java:
```diff
@@ -200,8 +214,16 @@ public ActivityLogEntry(AccountAttributes userAccount, boolean isMasquerade, Str
     
+    private String formatCurrentTimeForId() {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmssSS");
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
```
I think you can use `sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));` here to get the admin time zone instead of using "UTC"

kanghj commented:
Changes made.
